src/UI/FlatList.re

```diff
@@ -4,106 +4,207 @@
  * Virtualized list helper
  */
 
+open Revery;
 open Revery.UI;
+open Revery_UI_Components;
 
-type renderFunction = int => React.syntheticElement;
-
-let component = React.component("FlatList");
+module Utility = Oni_Core.Utility;
 
-let additionalRowsToRender = 1;
+// TODO: Remove after 4.08 upgrade
+module Option = {
+  let map = f =>
+    fun
+    | Some(x) => Some(f(x))
+    | None => None;
 
-type glRenderFunction = (int, float) => unit;
+  let value = (~default) =>
+    fun
+    | Some(x) => x
+    | None => default;
 
-let render =
-    (~scrollY=0., ~rowHeight, ~height, ~count, ~render: glRenderFunction, ()) => {
-  let rowsToRender = rowHeight > 0. ? int_of_float(height /. rowHeight) : 0;
-  let startRowOffset =
-    rowHeight > 0. ? int_of_float(scrollY /. rowHeight) : 0;
-  let pixelOffset = mod_float(scrollY, rowHeight);
+  let some = x => Some(x);
+};
 
-  let i = ref(max(startRowOffset - additionalRowsToRender, 0));
+type renderFunction = int => React.syntheticElement;
 
-  let len = count;
+module Constants = {
+  let scrollWheelMultiplier = 25;
+  let additionalRowsToRender = 1;
+  let scrollBarThickness = 6;
+  let scrollTrackColor = Color.rgba(0.0, 0.0, 0.0, 0.4);
+  let scrollThumbColor = Color.rgba(0.5, 0.5, 0.5, 0.4);
+};
 
-  while (i^ < rowsToRender
-         + additionalRowsToRender
-         + startRowOffset
-         && i^ < len) {
-    let item = i^;
-    let rowOffset = float_of_int(item - startRowOffset) *. rowHeight;
+module Styles = {
+  let container = (~width as w, ~height as h) =>
+    Style.[
+      position(`Relative),
+      top(0),
+      left(0),
+      width(w),
+      height(h),
+      overflow(`Hidden),
+    ];
+
+  let slider =
+    Style.[
+      position(`Absolute),
+      right(0),
+      top(0),
+      bottom(0),
+      width(Constants.scrollBarThickness),
+    ];
+
+  let viewport = (~isScrollbarVisible) =>
+    Style.[
+      position(`Absolute),
+      top(0),
+      left(0),
+      right(isScrollbarVisible ? 0 : Constants.scrollBarThickness),
```

Comment:
I spent a little time troubleshooting the 'hover' issue - if we put `bottom(0)` here, it seems to fix it. It looks like this was treated as having a smaller height than expected, causing it and the children to not get the expected mouse events.

Reply:
Huh, too simple... But awesome!